src/connector/src/source/kafka/split.rs

```diff
@@ -24,6 +24,12 @@ pub struct KafkaSplit {
     pub(crate) partition: i32,
     pub(crate) start_offset: Option<i64>,
     pub(crate) stop_offset: Option<i64>,
+    #[serde(skip)]
+    /// Used by shared source to hackily seek to the latest offset without fetching start offset first.
+    /// XXX: But why do we fetch low watermark for latest start offset..?
+    ///
+    /// When this is `true`, `start_offset` will be ignored.
+    pub(crate) hack_seek_to_latest: bool,
```

Comment:
suggest changing another name like `force_start_latest`. This name sounds like doing some magic 🪄

Reply:
I do think this is hacky and magical.. If possibile, I want to only use `start_offset` to do this. In other words, also use `Offset::End` for `scan.startup.mode='latest'`